### PR TITLE
Remove protoarray forkchoice

### PR DESCRIPTION
**Pull request by potuz**
Removes protoarray implementation of forkchoice.

### CODE REVIEW

File: beacon-chain/blockchain/process_block_helpers.go [hunk 1]
```diff
@@ -163,7 +162,7 @@ func (s *Service) updateFinalized(ctx context.Context, cp *ethpb.Checkpoint) err
 
 	fRoot := bytesutil.ToBytes32(cp.Root)
 	optimistic, err := s.cfg.ForkChoiceStore.IsOptimistic(fRoot)
-	if err != nil && err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
+	if err != nil && err != doublylinkedtree.ErrNilNode {
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/process_block_helpers.go:165]: errors.Is is the preferable approach to checking errors in this manner.

```suggestion
	if err != nil && !errors.Is(err, doublylinkedtree.ErrNilNode) {
```

https://pkg.go.dev/errors#pkg-overview

File: beacon-chain/blockchain/chain_info.go [hunk 2]
```diff
@@ -314,7 +313,7 @@ func (s *Service) IsOptimistic(ctx context.Context) (bool, error) {
 	if err == nil {
 		return optimistic, nil
 	}
-	if err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
+	if err != doublylinkedtree.ErrNilNode {
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/chain_info.go:316]: errors.Is is the preferable approach to checking errors in this manner.

https://pkg.go.dev/errors#pkg-overview




```suggestion
	if !errors.Is(err, doublylinkedtree.ErrNilNode) {
```

File: beacon-chain/blockchain/chain_info.go [hunk 3]
```diff
@@ -338,7 +337,7 @@ func (s *Service) IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool,
...
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/chain_info.go:340]: errors.Is is the preferable approach to checking errors in this manner.

https://pkg.go.dev/errors#pkg-overview


```suggestion
	if !errors.Is(err, doublylinkedtree.ErrNilNode) {
```

File: beacon-chain/forkchoice/doubly-linked-tree/store.go [hunk 4]
```diff
@@ -261,13 +261,19 @@ func (s *Store) tips() ([][32]byte, []types.Slot) {
 func (f *ForkChoice) HighestReceivedBlockSlot() types.Slot {
 	f.store.nodesLock.RLock()
 	defer f.store.nodesLock.RUnlock()
+	if f.store.highestReceivedNode == nil {
+		return 0
+	}
```
Comment (prestonvanloon) [inline beacon-chain/forkchoice/doubly-linked-tree/store.go:266]: Nice check :+1:

**Review by terencechain — COMMENTED** [review]
Lgtm

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**